azext_edge/edge/providers/orchestration/deletion.py

```diff
@@ -0,0 +1,194 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from sys import maxsize
+from time import sleep
+from typing import TYPE_CHECKING, Dict, List, Optional, Tuple
+
+from knack.log import get_logger
+from rich import print
+from rich.console import NewLine
+from rich.live import Live
+from rich.progress import Progress, SpinnerColumn, TimeElapsedColumn
+from rich.prompt import Confirm
+from rich.table import Table
+
+from ...util.az_client import get_resource_client, wait_for_terminal_states
+from .resource_map import IoTOperationsResource, IoTOperationsResourceMap
+
+logger = get_logger(__name__)
+
+
+if TYPE_CHECKING:
+    from azure.core.polling import LROPoller
+
+
+def delete_ops_resources(
+    cmd,
+    cluster_name: str,
+    resource_group_name: str,
+    confirm_yes: Optional[bool] = None,
+    no_progress: Optional[bool] = None,
+    force: Optional[bool] = None,
+):
+    manager = DeletionManager(
+        cmd=cmd, cluster_name=cluster_name, resource_group_name=resource_group_name, no_progress=no_progress
+    )
+    manager.do_work(confirm_yes=confirm_yes, force=force)
+
+
+class DeletionManager:
+    def __init__(
+        self,
+        cmd,
+        cluster_name: str,
+        resource_group_name: str,
+        no_progress: Optional[bool] = None,
+    ):
+        from azure.cli.core.commands.client_factory import get_subscription_id
+
+        self.cmd = cmd
+        self.cluster_name = cluster_name
+        self.resource_group_name = resource_group_name
+        self.resource_map = IoTOperationsResourceMap(
+            cmd=cmd, cluster_name=cluster_name, resource_group_name=resource_group_name
+        )
+        self.subscription_id = get_subscription_id(cli_ctx=cmd.cli_ctx)
+        self.resource_client = get_resource_client(self.subscription_id)
+
+        self._render_progress = not no_progress
+        self._live = Live(None, transient=False, refresh_per_second=8, auto_refresh=self._render_progress)
+        self._progress_bar = Progress(
+            SpinnerColumn(),
+            *Progress.get_default_columns(),
+            "Elapsed:",
+            TimeElapsedColumn(),
+            transient=False,
+        )
+        self._progress_shown = False
+
+    def do_work(self, confirm_yes: Optional[bool] = None, force: Optional[bool] = None):
+        self._display_resource_tree()
+
+        should_delete = True
+        if not confirm_yes:
+            should_delete = Confirm.ask("Continue?")
+
+        if not should_delete:
```

Comment:
if (not confirm_yes) or Confirm.ask("Continue?"):

Reply:
I think it is easier to read as it is now.